database/factories/ReciterFactory.php

```diff
@@ -5,14 +5,13 @@
 use Faker\Generator as Faker;
 
 $factory->define(Reciter::class, function (Faker $faker) {
+	$user = factory(User::class)->create();
     return [
         'name' => $name = $faker->name,
         'slug' => str_slug($name),
         'description' => $faker->paragraph,
         'avatar' => $faker->imageUrl(640, 480, 'people'),
         'status' => 0,
-        'created_by' => function () {
-            return factory(User::class)->create()->id;
-        },
+        'created_by' => User::all()->random()->id,
```

Comment:
Why did you change this? This was done properly before, and now this version of it is less correct.

Reply:
the reason why i changed this was because currently you are creating a new user every time you run this factory, and personally i thought that was not how i thought it should be, i thought that selecting an existing user was more realistic

Reply:
The way the `created_by` attribute was previously defined, we would create a new user if you just called the factory like this:

```php
$reciter = factory(Reciter::class)->create();
```

But if you wanted to specify which user ID, you could do that too...
```php
$user = User::find($userId);
$reciter = factory(Reciter::class)->create(['created_by' => $user->id]);
```

Now, you have it so that it will just select a random user ID. And if you haven't made sure to create a user before calling the factory, you'll see an exception.